python/llm/test/inference_gpu/test_optimize_model.py

```diff
@@ -54,6 +58,126 @@ def test_optimize_model(Model, Tokenizer, model_path):
     diff = abs(logits_base_model - logits_optimized_model).flatten()
 
     assert any(diff) is False
+"""
+
+class Test_Optimize_Gpu_Model:
+    def setup(self):
+
+        self.layer_inputs = []
+        self.layer_outputs = []
+        self.pre_layer_outputs = []
+
+    def run_optimize_gpu_model(self, Model, Tokenizer, model_path, self_attn, prev_attn, lower_bound):
+        def forward_hook(module, input, output, layer_name):
+            self.layer_outputs.append(output)
+
+        def pre_forward_hook(module, input, output, layer_name):
+            self.pre_layer_outputs.append(output)
+
+        def new_forward_hook(module, input):
+            if model_path == os.environ.get('LLAMA2_7B_ORIGIN_PATH'):
+                repalcement_norm = model.model.norm
+                opt_model.model.norm = repalcement_norm
```

Comment:
I don't think it is a good practice to write code like this, since opt_model is not even defined here.